src/test/store/js-tracer.test.js

```diff
@@ -0,0 +1,170 @@
+        'Interpreter (2:4) | IonMonkey (6:18)',
+      ]);
+    });
+  });
+  describe('self time', function() {
+    it('has empty JS tracer timing if no events are in the js tracer table', function() {
+      expect(
+        getHumanReadableJsTracerTiming({ selfTime: true, events: [] })
+      ).toEqual([]);
+    });
+
+    it('can generate some simple nested timing', function() {
+      expect(
+        getHumanReadableJsTracerTiming({
+          selfTime: true,
+          events: [
+            ['https://mozilla.org', 0, 20],
+            ['Interpreter', 1, 19],
+            ['IonMonkey', 2, 18],
+          ],
+        })
+      ).toEqual([
+        'Interpreter (1:2) | Interpreter (18:19)',
+        'IonMonkey (2:18)',
+        'https://mozilla.org (0:1) | https://mozilla.org (19:20)',
+      ]);
+    });
+
+    it('can "drain off" prefixes the first if branch of the algorithm', function() {
+      // This test is checking a specific if branch of the timing function:
+      //
+      //    xxxxxxxxxxxxxxxx[================]
+      //    xxxxxxxx[======]     [current]
+      //    [prefix]
+      expect(
+        getHumanReadableJsTracerTiming({
+          selfTime: true,
+          events: [
+            // Make the formatting "prettier".
+            ['A', 0, 2],
+            ['B', 0, 2],
+            ['C', 0, 1],
+            ['A', 2, 3],
+          ],
+        })
+      ).toEqual([
+        // Make the formatting "prettier".
+        'A (2:3)',
+        'B (1:2)',
+        'C (0:1)',
+      ]);
+    });
+
+    it('can split a prefix stack if the ending time matches', function() {
+      // This test is checking a specific if branch of the timing function:
+      //
+      //   [prefix]xxxxxxxxx
+      //           [current]
+      expect(
+        getHumanReadableJsTracerTiming({
+          selfTime: true,
+          events: [
+            // Make the formatting "prettier".
+            ['A', 0, 2],
+            ['B', 1, 2],
+          ],
+        })
+      ).toEqual([
+        // Make the formatting "prettier".
+        'A (0:1)',
+        'B (1:2)',
+      ]);
+    });
+  });
+});
+
+/**
+ * Create JS tracing information from a list of events, and then compute a human
+ * readable representation of the timing.
+ *
+ * "EventName1 (StartTime:EndTime) | EventName2 (StartTime:EndTime)"
+ */
+function getHumanReadableJsTracerTiming({
+  selfTime,
+  events,
+}: {
+  selfTime: boolean,
+  events: TestDefinedJsTracerEvent[],
+}): string[] {
+  const profile = getProfileWithJsTracerEvents(events);
+  const { dispatch, getState } = storeWithProfile(profile);
+  dispatch(changeShowJsTracerSummary(selfTime));
+
+  return ensureExists(
+    selectedThreadSelectors.getExpensiveJsTracerTiming(getState())
+  ).map(({ start, end, label, length }) => {
+    const lines = [];
+    for (let i = 0; i < length; i++) {
+      lines.push(
+        `${label[i]} (${parseFloat(start[i].toFixed(2))}:${parseFloat(
+          end[i].toFixed(2)
+        )})`
```

Comment:
Why do you use `parseFloat` here? Isn't the use of `toFixed` enough?

Reply:
ah I see. For tests I wouldn't mind having extra 0 :-) but OK 

This is something that toLocaleString (that we used in format-numbers.js recently) helps with, but obviously we don't want our test to depends on the locale...